phoenix-core/src/main/java/org/apache/phoenix/schema/types/PUnsignedLong.java

```diff
@@ -95,6 +95,11 @@ public Object toObject(byte[] b, int o, int l, PDataType actualType, SortOrder s
   }
 
   @Override
+    public boolean isCastableTo(PDataType targetType) {
```

Comment:
Was thinking to only make this change so that an UNSIGNED_LONG is castable to a TIMESTAMP. Allowing an INTEGER to be cast to a TIMESTAMP seems weird to me. How about we just start with this change for now and you adjust your tests accordingly?